tips/TIP-0045/tip-0045.md

```diff
@@ -0,0 +1,1611 @@
+- Return `Size Score + Input Score + Context Input Score + Output Score + Allotment Score + Signature Score`.
+
+## Validation
+
+A <i>Transaction Payload</i> has different validation stages since some validation steps can only be executed when
+certain information has (or has not) been received. We therefore distinguish between syntactic and semantic validation.
+
+The different output types and optional output features introduced by [TIP-38](../TIP-0038/tip-0038.md),
+[TIP-1](../TIP-0042/tip-0041.md),[TIP-42](../TIP-0042/tip-0042.md), [TIP-43](../TIP-0043/tip-0043.md) and
+[TIP-44](../TIP-0044/tip-0044.md) add additional constraints to the transaction validation rules, but since these are
+specific to the given outputs and features, they are discussed for each
+[output type](../TIP-0038/tip-0038.md#output-design) and [feature type](../TIP-0038/tip-0038.md#features) separately.
+
+### Syntactic validation
+
+Syntactic validation is checked as soon as the transaction has been received. It validates the structure but not the
+signatures of the transaction. If the transaction does not pass this stage, it must not be broadcast further and can be
+discarded right away.
+
+The following criteria defines whether a payload passes the syntactical validation:
+
+- Transaction:
+  - `Network ID` must match the value of the current network.
+  - `CreationTime` must be a valid `Slot index` timestamp.
+  - Inputs:
+    - `Inputs Count` must be `0 < x ≤ Max Inputs Count`.
+    - For each input the following must be true:
+      - `Input Type` must denote an allowed input type, listed in section [Inputs](#inputs).
+      - `Transaction Output Index` must be `0 ≤ x < Max Outputs Count`.
+    - Each pair of `Transaction ID` and `Transaction Output Index` must be unique in the list of inputs.
+  - Context Inputs:
+    - `Context Inputs Count` must be `0 ≤ x ≤ Max Inputs Count`.
+    - `Context Input Type` must denote an allowed input type, listed in section [Context Inputs](#context-inputs).
+    - There must be zero or one _Commitment Input_.
+    - All _Rewards Inputs_ must reference a different `Index` and it must hold that: `Index <= Max Inputs Count`.
+    - All _Block Issuance Credit Inputs_ must reference a different `Account ID`.
+  - Outputs:
+    - `Outputs Count` must be `0 < x ≤ Max Outputs Count`.
+    - For each output the following must be true:
+      - `Output Type` must match one of the values described under [Outputs](#outputs).
+      - The `Amount` field must not be `0`.
+      - The `Amount` field must be at least the minimum storage deposit according to [TIP-47](../TIP-0047/tip-0047.md).
+      - The output itself must pass syntactic validation.
+    - The sum of all `Amount` fields must not exceed `Max IOTA Supply`.
+    - The sum of all `Mana` fields must not exceed <code>2<sup>Mana Bits Count</sup> - 1</code>.
+    - The count of all distinct native tokens present in outputs must not be larger than `Max Native Token Count`.
```

Comment:
I think we said we would remove this limitation now since it's one token per output we're just bound to the number of ouput per tx

Reply:
Yeah and I can't find that check anymore so I think it was removed :+1:.